lib/factory.js

```diff
@@ -64,6 +64,13 @@ function create(options) {
       ' - Because of this, you should never use host and port options together');
   }
 
+  if (options.protocol) {
+    var protocolOptionWarning = new InvalidArgumentsError('The "protocol" option' +
+      ' does not affect socketcluster-client. If you want to utilize SSL/TLS' +
+      ' - use "secure" option instead');
+    console.warn(protocolOptionWarning);
```

Comment:
We should avoid using console.log or console.warn in this library because some people use custom loggers (especially when running the client from inside Node.js) and we don't want to spam stdout with our own logging.

We can either:

1. Throw the error (like all the errors above it) and let the user handle that error themselves; but I guess maybe this is too extreme in this case. I understand your point that maybe we don't want to crash the whole application in this case.

2. Inside `scclientsocket.js` when instantiating, you can pass the error to `self._onSCError(protocolOptionWarning)`; I believe that this will cause the 'error' event to be emitted on the client socket object but it shouldn't crash the application (assuming that the 'error' event is handled by the user's code); also in this case the socket should continue working normally even if the 'error' does trigger.